### PR TITLE
Fix `AliasMetaData#fromXContent` parsing

### PR DESCRIPTION
**Pull request by olcbean**
While working on #28799, it was detected that `AliasMetaData` is not correctly parsed. 
This PR fixes the issue and adds dedicated XContent tests as well.

### CODE REVIEW

**Comment by javanna** [comment]
test this please

File: server/src/test/java/org/elasticsearch/cluster/metadata/AliasMetaDataTests.java [hunk 1]
```diff
@@ -52,4 +53,49 @@ public void testSerialization() throws IOException {
...
+            parser.nextToken();
+        }
+        assertEquals(XContentParser.Token.FIELD_NAME, parser.currentToken());
+        AliasMetaData aliasMetaData = AliasMetaData.Builder.fromXContent(parser);
+        assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
+        return aliasMetaData;
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
```
Comment (javanna) [inline server/src/test/java/org/elasticsearch/cluster/metadata/AliasMetaDataTests.java]: could you add a comment that clarifies this is needed as this class is used by the high-level REST client. I am quite worried that this class is internal (like many others though) yet its parsing may be modified without taking the client into account.

**Review by javanna — APPROVED** [review]
left a tiny comment, LGTM though. thank you!

**Comment by javanna** [comment]
retest this please

File: server/src/test/java/org/elasticsearch/cluster/metadata/AliasMetaDataTests.java (hunk 1 above)
Comment (olcbean) [reply server/src/test/java/org/elasticsearch/cluster/metadata/AliasMetaDataTests.java]: @javanna I can add a comment later today or the commit msg is enough?
Sorry for getting back to you that late.

Comment (javanna) [reply server/src/test/java/org/elasticsearch/cluster/metadata/AliasMetaDataTests.java]: no worries at all, it was not that important. and I was not willing to wait for another test run after adding the comment addition so I merged :)

**Pull request merged**